### ISSUE TITLE
hw_disk_bus: scsi for kubernetes images

### ISSUE THREAD

**Issue by linwalth** [osism/openstack-image-manager#564]
I have found through some digging that the use of virtio as hw_disk_bus is not needed anymore.

(see https://github.com/osism/openstack-image-manager/issues/482)

so these parameters can be set to scsi (as scsi appears to be slightly more perfomant, allowing for DISCARD [fstrim].)

https://github.com/osism/openstack-image-manager/blob/main/etc/images/kubernetes.yml#L15

https://github.com/osism/openstack-image-manager/blob/main/etc/images/gardenlinux.yml#L14

**Comment by artificial-intelligence** [comment]
I'd be really hesitant to switch the default here and would like to see actual current numbers that scsi is more performant.

as said elsewhere already: virtio-blk does support discard.

**Comment by artificial-intelligence** [comment]
So, i quickly re-researched the performance topic and so far I mostly found sources in favor of virtio-blk when it comes to performance:

- https://www.qemu.org/2021/01/19/virtio-blk-scsi-configuration/ says virtio-blk is faster than scsi (the test cited is random read 4k)
- the above source also notes, that scsi passthrough was disabled in the 5.6 kernel and replaced by virtio-scsi
- https://stackoverflow.com/a/45487683 mentions also, like the above source, that virtio-scsi is more complex code and thus slower; but it also contains the old information that virtio-scsi does support TRIM, which virtio-blk also supports.
- finally there is this [blog from 2017](https://mpolednik.github.io/2017/01/23/virtio-blk-vs-virtio-scsi/) which has the most datapoints and tests running fio. most datapoints here are in favor of virtio-blk speed as well.

**Comment by berendt** [comment]
Please take to an SCS meeting an discuss there.

**Comment by artificial-intelligence** [comment]
@linwalth can you check if virtio-blk works for you in terms of discard? do you see any performance difference - positive or negative - for your workload?

I don't know your platform/usecase, as I'm still a little new here around :smile: 

If you have a problem with the current value or if you can show data that scsi works better for your usecase, we could discuss switching this again.

Thanks!

**Comment by artificial-intelligence** [comment]
@berendt I added the topic to the agenda, I hope it's okay this way:

https://input.scs.community/#2023-05-17

**Comment by fkr** [comment]
> https://input.scs.community/#2023-05-17

that link is kaput ;)

Where did you add it?

**Comment by artificial-intelligence** [comment]
yeah, the markdownlinks are somehow defect, it's here (deeplinks seem to be broken somehow):

https://input.scs.community/2023-scs-team-iaas

**Comment by linwalth** [comment]
> @linwalth can you check if virtio-blk works for you in terms of discard? do you see any performance difference - positive or negative - for your workload?

I was just passing this issue down, I have no immediate contact with the usecase itself.
